docs/training-run/index.txt

```diff
@@ -4,106 +4,269 @@
 
-Our library automatically communicates with the master to get what needs to run and report back the
-results. It might:
+**********************
+ Create an Experiment
+**********************
+
+Currently, the CLI is the only supported mechanism for creating an experiment, although, you can
+create an experiment from an existing experiment or trial using the WebUI. To create an experiment:
+
+.. code::
+
+   $ det experiment create <configuration file> <context directory>
+
+-  The :ref:`Experiment Configuration <experiment-configuration>` file is a YAML file that controls
+   your experiment.
+-  The context directory contains relevant training code, which is uploaded to the master.
+
+The total size of the files in the context cannot exceed 95 MB. As a result, only very small
+datasets should be included. Instead, set up data loaders to read data from an external source.
+Refer to the :ref:`Prepare Data <prepare-data>` section for more data loading options.
+
+Because project directories can include large artifacts that should not be packaged as part of the
+model definition, including data sets or compiled binaries, users can specify a ``.detignore`` file
+at the top level, which lists the file paths to be omitted from the model definition. The
+``.detignore`` file uses the same syntax as `.gitignore <https://git-scm.com/docs/gitignore>`__.
+Byte-compiled Python files, including ``.pyc`` files and ``__pycache__`` directories, are always
+ignored.
+
+********************
+ Pre-training Setup
+********************
+
+Trials are created to train the model. The :ref:`Hyperparameter Tuning <hyperparameter-tuning>`
+searcher specified in the experiment configuration file defines a set of hyperparameter
+configurations. Each set corresponds to a single trial.
+
+After the context and experiment configuration reach the master, the experiment waits for the
+scheduler to assign slots. If there are insufficient idle slots available, the master automatically
+scales up the resource pools.
+
+When a trial is ready to run, the master communicates with the agent, or :ref:`distributed training
+<multi-gpu-training>` agents, which create(s) containers that have the configured environment and
+training code. A set of default container images applicable to many deep learning tasks is provided,
+but you can also specify a :ref:`custom image <custom-docker-images>`. If the specified container
+images do not exist locally, the trial container fetches the images from the registry.
+
+After starting the containers, each trial runs the ``startup-hook.sh`` script in the context
+directory.
 
-#. train the model for a few batches on the training dataset;
-#. checkpoint the states of the model and other objects;
-#. validate the model on the validation dataset.
+The pre-training activity can incur a delay before each trial begins training but typically only
+takes a few seconds.
```

Comment:
there was some discussion on this point before.  An earlier version said something like, "this might take several minutes" and I was like, "don't make us look bad, when often it only takes a couple seconds".

I personally think any step which might take multiple minutes of waiting is worth documenting why it takes that long, but also worth mentioning that it can be nearly instantaneous under the right conditions (aka cloud spinup delays are not our fault).

Reply:
Sounds good as is, and the user will not be surprised if it does take long.